sys/kern/exec.c

```diff
@@ -302,13 +303,32 @@ static void destroy_vmspace(exec_vmspace_t *saved) {
   vm_map_delete(saved->uspace);
 }
 
+static bool check_setid(vnode_t *vn, uid_t *uid, gid_t *gid) {
+  vattr_t attr;
+  *uid = *gid = -1;
+
+  if (VOP_GETATTR(vn, &attr))
+    return false;
+
+  if (attr.va_mode & S_ISUID)
+    *uid = attr.va_uid;
+
+  if (attr.va_mode & S_ISGID)
+    *uid = attr.va_gid;
```

Comment:
Shouldn't be here `*gid` instead of `*uid`?

Reply:
Thanks. It should.